src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs

```diff
@@ -69,6 +69,7 @@ private async Task<ProjectStateChecksums> ComputeChecksumsAsync(CancellationToke
                     var analyzerConfigDocumentChecksums = new ChecksumCollection(await analyzerConfigDocumentChecksumTasks.WhenAll().ConfigureAwait(false));
 
                     return new ProjectStateChecksums(
+                        this.Id,
```

Comment:
>this.Id, [](http://example.com/codeflow?start=24&length=8)

General question about something not changed in this PR:

I see SelectAsArray done above, which I understand gives back an ImmutableArray that allows struct based enumeration. However, that array did require a heap allocation. It seems to me that if you know the collection will only be enumerated once locally (as in this case), it's better to have a heap based enumerator allocated than an ImmutableArray of a potentially large size.

Of course, that assumption could be broken and then you end up doing multiple enumerator allocations, but that doesn't seem too likely and too costly if it does happen. Another option if repeated enumerations is a concern would be an ArrayBuilder based SelectAs method, as that could have the benefit of no-allocation and struct based enumeration.

Or maybe I'm just overthinking this.